aten/src/ATen/core/jit_type.h

```diff
@@ -661,6 +661,17 @@ struct CAFFE2_API TensorType : public Type {
     return scalar_type_ && device_ && sizes_.isComplete() && strides_.isComplete();
   }
 
+  bool isInferredType() const {
+    return is_inferred_;
+  }
+
+  static TensorTypePtr getInferred() {
```

Comment:
This function will return a new instance every time it is called. I think we should use a singleton here to avoid that and reuse the same instance across calls. See `TensorType::get` for an example. As we discussed, you can modify or create and overload of `TensorType::create` if needed.